Nixtla/statsforecast
# Add Mintlify automations

## Summary
This PR adds the automations necessary to create the statsforecast Mintlify docs. When a push to `main` is made, the `build-docs` workflow will run which will upload the needed Mintlify docs files at the `docs` branch.

### Steps needed before merging
We should merge this when the Mintlify docs are ready for all 4 repos (I will be making similar PRs on all of them).
When we merge this, somebody on the Nixtla team should also install the Mintlify GitHub App which I will provide a link to inside the #mintlify-nixtla Slack channel.

- [ ] Install mintlify github app
- [ ] Prepare mintlify for all the other repos
- [ ] [Configure the default GITHUB_TOKEN permissions](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#configuring-the-default-github_token-permissions) by selecting read and write permissions - this is so that the `build-docs` script works.

## Maintenance instructions
When a new page is added it needs to be added to `mint.json`. We can automate this too on the Mintlify side (Similar to the setup you guys have right now with the `sidebar.yml`) but this will take a couple weeks on our end.

## Test plan
- Install the [Mintlify CLI](https://www.npmjs.com/package/mintlify) (Prerequisite: [node.js](https://nodejs.org/en/download), if you have a mac I like to use [nvm](https://github.com/nvm-sh/nvm)
- Run `bash ./preview-mintlify-docs.bash` to get a preview of the docs locally. Note - the preview will only show `statsforecast` but the Nixtla deployment will be a frankenstein merged version of all 4 docs instances. I created this command so that developers can preview & sanity check their changes before pushing.
